cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -246,9 +246,9 @@ where
 					};
 
 					if let Err(e) =
-						self.execution_receipt_sender.unbounded_send(execution_receipt.clone())
+						self.execution_receipt_sender.unbounded_send(signed_execution_receipt)
 					{
-						tracing::error!(target: LOG_TARGET, error = ?e, "Failed to send execution receipt");
+						tracing::error!(target: LOG_TARGET, error = ?e, "Failed to send signed execution receipt");
 					}
 
 					// Return `Some(_)` to broadcast ER to all farmers via unsigned extrinsic.
```

Comment:
Why below ER is not signed even though comment says it needs to be broadcast to other executors, isn't it what the channel above does?

Reply:
The comment said it will be broadcasted to all **farmers** via unsigned extrinsic, farmers will see `ExecutionReceipt` from the unsigned extrinsic, executors will see `SignedExecutionReceipt` sent over the channel from the executor gossip network.

Reply:
But farmers also need to have signed extrinsic, don't they?

Reply:
I don't understand why farmers need to have signed extrinsic. We have no signed extrinsic on this topic so far.

Reply:
How would farmers know that ER is produced by allowed executor and not by some random person on the Internet?

Reply:
That's one thing I'd love to discuss along with the above TODO of execution receipt signing misbehavior handling. Farmers can not verify the validity of the execution receipt signature of a previous block within the runtime because runtime can not access the info of some old block. Another TODO :P https://github.com/subspace/subspace/blob/e848e9e153e84fa90c6e5eddd1cfabee63cdece6/crates/pallet-executor/src/lib.rs#L85

Reply:
Farmers should be able to verify validity because they have access to executor set and signature must be produced by one of executors that are designated to produce ER at this block height.

Reply:
Especially now when we have just one executor

Reply:
Well, it should verify the validity for sure, the problem is how we implement it, as I mentioned, this verification can't be done in the runtime directly, which is similar to the fraud proof verification problem.

Reply:
We definitely can't verify the contents, but we don't need that anyway unless we have a fraud proof. But we can and need to verify the signature so at least we know it is produced by properly staked executor.

Reply:
As I have said, you can't do it in runtime directly but have to verify it on the client side as you can't access the executor set of a previous block within the runtime(the execution receipt belongs to a previous block). Then the problem becomes if we're going to use the externality approach or a complete off-runtime approach. Personally, I second the externality way as this verification is relatively effortless.

Reply:
Now I see the problem better, thanks for clarifying. I don't think this is unsolvable though. If we update validator set at the end of the block (after checking all of the extrinsics, including ERs), we don't need to reach out to the previous block at all. Also we can keep 2 sets in the state potentially.

I think this is very important and needs to change (even if we don't validate the transaction in the runtime, it still needs to be signed so the client can check it), but we can postpone it to the next PR.

Reply:
The previous block does not mean the just last block, possibly some old block which is a few blocks earlier as you can't guarantee the receipt of the current block will be included in the next block? Additionally, if we have another way which is not complicated in my opinion, why do we have to unnecessarily add more state in the runtime?

Yeah, we'll discuss a few things before moving forward.